### ISSUE TITLE
Eliminate successful tests

### ISSUE THREAD

**Issue by Naatan** [haveyoudebuggedit/gotestfmt#1]
Hi there,

First of all; GREAT tool! It makes grokking results so much easier.

One shortcoming of the tool is that you have to run tests with `-v`, which makes the output very verbose. Let's say you have 500 tests and the one that's failing is number 380, that's 380 entries you have to scroll through until you find the failure you're interested in.

It would be nice if there was an option to simply dump all the successful tests. There's no reason for most people to look at those.

**Comment by janosdebugs** [comment]
Hey @Naatan,

My apologies for the late answer, I didn't get a notification about your issue. I agree, this would be a great option. I'm also considering if it would be worth adding an option to put the failing tests first.

I'll spend some time on this and release an update to add this option.

**Comment by janosdebugs** [comment]
@Naatan I after looking at the code I just realized that this is already supported by changing the rendering template. You can create your own rendering template in a `.gotestfmt` directory and add the two files from the [repository](https://github.com/haveyoudebuggedit/gotestfmt/tree/main/.gotestfmt). You can then change the `package.gotpl` file to read as follows:

```
{{- /*gotype: github.com/haveyoudebuggedit/gotestfmt/parser.Package*/ -}}
{{- /*
This template contains the format for an individual package. GitHub actions does not currently support nested groups so
we are creating a stylized header for each package.
*/ -}}
{{- if eq .Result "PASS" -}}
    {{ "\033" }}[0;32m
{{- else if eq .Result "SKIP" -}}
    {{ "\033" }}[0;33m
{{- else -}}
    {{ "\033" }}[0;31m
{{- end -}}
📦 {{ .Name }}{{ "\033" }}[0m{{- "\n" -}}
{{- with .Reason -}}
    🛑 {{ . -}}{{- "\n" -}}
{{- end -}}
{{- with .Output -}}
    {{- . -}}{{- "\n" -}}
{{- end -}}
{{- with .TestCases -}}
    {{- $failed := false -}}
    {{- range . -}}
        {{- if ne .Result "PASS" -}}
            {{- $failed = true -}}
            ::group::
            {{- if eq .Result "SKIP" -}}
                {{ "\033" }}[0;33m🚧
            {{- else -}}
                {{ "\033" }}[0;31m❌
            {{- end -}}
            {{ " " }}{{- .Name -}}
            {{- "\033" -}}[0;37m ({{ .Duration -}}
            {{- with .Coverage -}}
                , coverage: {{ . }}%
            {{- end -}})
            {{- "\033" -}}[0m
            {{- "\n" -}}

            {{- with .Output -}}
                {{- . -}}
                {{- "\n" -}}
            {{- end -}}

            ::endgroup::{{- "\n" -}}
        {{- end -}}
    {{- end -}}
    {{- if not $failed -}}
        {{ "\033" }}[0;32m✅ All tests passed{{- "\033" -}}[0m
    {{- end -}}
{{- end -}}
{{- "\n" -}}
```

I will still keep this open as I think it would be good to have an easier way to do this, but the template above should help you already.

**Comment by Naatan** [comment]
That's awesome, thanks @janosdebugs I'll give that a try!

I'm happy to use this solution, though I can definitely see why making this easier would be helpful.

**Comment by Naatan** [comment]
Here's the template I ended up using. The one above was still a little too verbose.

https://github.com/ActiveState/cli/pull/1626/files#diff-ebd8dc03fcdbac7ddf1b8c86e69aa20785f5a8940de37330c4488fb6e1d3502b

**Comment by janosdebugs** [comment]
Perfect! The customization was worth it after all. ;)

**Comment by janosdebugs** [comment]
Hey @Naatan as a bugfix for #2, I've added support for running `go test` without `-v`. Could you try and see if these binaries work for you to get a shorter output? https://github.com/haveyoudebuggedit/gotestfmt/actions/runs/1448047829

**Comment by janosdebugs** [comment]
The fix for #2 has just been released as version 1.0.2.

**Comment by janosdebugs** [comment]
@Naatan closing this as running without `-v` is now supported and should yield the results you were looking for. Thanks again for the feedback!
